### DISCUSSION TITLE
[defaults]: Consider Reducing Default max_htlcx_accepted

### DISCUSSION THREAD

**Discussion by carlaKC** [lightningnetwork/lnd#10241] [Ideas]
https://github.com/bitcoin/bitcoin/pull/29873 proposes a 10,000 vbyte limit on unconfirmed parent TRUC(V3) transactions, which in the context of lightning would be our commitment transaction. Given our expected commitment weight in [bolt-03](https://github.com/lightning/bolts/blob/master/03-transactions.md#expected-weight-of-the-commitment-transaction) for anchor channels we can calculate the maximum HTLCs that would fit in this transaction limit: 
```
(900 + 172 * num-htlc-outputs + 224)/4 = 10000 vbytes
num-htlc-outputs = 226
```

Divided between two channel participants we arrive at `max-accepted-htlcs = 113`, which is below LND's current limit of [483](https://github.com/lightningnetwork/lnd/blob/4417229bc800752691217c27a7ce615e98c94135/config.go#L174). While we'll definitely need dynamic commitments to migrate legacy channels to set a lower limit, upgrading the default now is a step in this direction for _new_  channels.

Opened this PR to start the conversation about the change of defaults in the context of LND as a project. If there's feedback on this limit, I think it makes sense to have the conversation on the bitcoin PR - as they're actively seeking input from LN folks.

**Comment by carlaKC** [comment]
I recognize that it seems somewhat unhinged for the person who's always yakking on about channel jamming to be talking about _decreasing_ the number of available slots on channels, but from all the data that we've collected so far it's apparent that we get nowhere near to these limits irl. Further, the expense of filling up these slots is so trivial for an attacker, that larger values don't really offer us much protection against jamming in practice anyway.

**Comment by MPins** [comment]
As the bitcoin/bitcoin#29873 was merged 4 days ago, it seems we should change the max-accepted-htlcs to 113 to keep compatibility, other way the commitment transactions should be refused in the near future!

**Comment by carlaKC** [comment]
> As the https://github.com/bitcoin/bitcoin/pull/29873 was merged 4 days ago, it seems we should change the max-accepted-htlcs to 113 to keep compatibility, other way the commitment transactions should be refused in the near future!

That's not the case, this will only come into play when we upgrade to using commitment transactions that opt in with `nVersion=3`.

**Comment by morehouse** [comment]
Concept ACK.

There are also good security reasons to have a smaller default and limit HTLC exposure.

**Comment by ziggie1984** [comment]
Concept ACK, now the only question to solve is how low do we choose the default value, I propose a similar value what eclair is using [30](https://github.com/ACINQ/eclair/blob/23c139ce91f3d911ca4512df342d56f4d40f80b0/eclair-core/src/main/resources/reference.conf#L125).

**Comment by morehouse** [comment]
> I propose a similar value what eclair is using [30](https://github.com/ACINQ/eclair/blob/23c139ce91f3d911ca4512df342d56f4d40f80b0/eclair-core/src/main/resources/reference.conf#L125).

SGTM

Some more data points:

- [LDK default is 50](https://github.com/lightningdevkit/rust-lightning/blob/1b281f15cc9b2f1e3deaf82e0f823ae4e7fd1df5/lightning/src/util/config.rs#L218)
- [CLN default is 30](https://github.com/ElementsProject/lightning/blob/6bcc05ac343a9915661bb187310149bda35e4abb/lightningd/options.c#L1102)

**Comment by carlaKC** [comment]
Something in the range of 30-50 sgtm, the nodes that have given us data for channel jamming get nowhere near that number of in-flight, and it puts us safely in the range for V3 txns.

**Comment by ziggie1984** [comment]
cc @saubyk let's prio it for LND 20

**Comment by saubyk** [comment]
cc: @Roasbeef for his take

**Comment by saubyk** [comment]
Converting to a discussion. Issue can be opened when taken up for development.
